src/Accelerators/NNPA/Runtime/zDNNExtension/zDNNExtension.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------------------ zDNNExtension.c -----------------------------===//
+//
+// Copyright 2024 The IBM Research Authors.
+//
+// =============================================================================
+//
+// Sets of extensions to the zdnn library.
+//
+//===----------------------------------------------------------------------===//
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "src/Accelerators/NNPA/Runtime/zDNNExtension/zDNNExtension.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+uint32_t ZTensorSplitSizeFromEnv() {
+  uint32_t cs = DEFAULT_ZTENSOR_SPLIT_SIZE;
+  const char *s = getenv("OM_ZTENSOR_SPLIT_SIZE");
+  if (s)
+    cs = atoi(s);
+  assert(cs % AIU_STICKS_PER_PAGE == 0 && "Chunk size is not multiple of 32");
+  if (ZTensorSplitDebugFromEnv())
+    printf("OM_ZTENSOR_SPLIT_SIZE: %d\n", cs);
+  return cs;
+}
+
+bool ZTensorSplitEnabledFromEnv() {
+  int enabled = DEFAULT_ZTENSOR_SPLIT_ENABLED;
+  const char *s = getenv("OM_ZTENSOR_SPLIT_ENABLED");
+  if (s)
+    enabled = atoi(s);
+  if (ZTensorSplitDebugFromEnv())
+    printf("OM_ZTENSOR_SPLIT_ENABLED: %d\n", enabled);
+  return (enabled != 0);
+}
+
+bool ZTensorSplitDebugFromEnv() {
+  int enabled = DEFAULT_ZTENSOR_SPLIT_DEBUG;
+  const char *s = getenv("OM_ZTENSOR_SPLIT_DEBUG");
+  if (s)
+    enabled = atoi(s);
+  return (enabled != 0);
+}
+
+void getZTensorShape(const zdnn_ztensor *t, zTensorShape *shape) {
+  // Stickification: (e4, e3, e2, e1) -> (e4, e1/64, e3, e2/32, 32, 64)
+  const zdnn_tensor_desc *desc = t->transformed_desc;
+  shape->dim6 = desc->dim4;
+  shape->dim5 = CEIL(desc->dim1, AIU_2BYTE_CELLS_PER_STICK);
+  shape->dim4 = desc->dim3;
+  shape->dim3 = CEIL(desc->dim2, AIU_STICKS_PER_PAGE);
+  shape->dim2 = AIU_STICKS_PER_PAGE;
+  shape->dim1 = AIU_2BYTE_CELLS_PER_STICK;
+  // Tensor size is large, use uint64_t.
+  uint64_t sizeFromDim = (uint64_t)shape->dim6 * (uint64_t)shape->dim5 *
+                         (uint64_t)shape->dim4 * (uint64_t)shape->dim3 *
+                         (uint64_t)shape->dim2 * (uint64_t)shape->dim1 *
+                         (uint64_t)AIU_2BYTE_CELL_SIZE;
+  uint64_t sizeFromBuffer = t->buffer_size;
+  assert(sizeFromDim == sizeFromBuffer && "buffer size mismatched");
+}
+
+zdnn_status allocZTensorChunk(const zdnn_ztensor *input, uint32_t axis,
+    uint32_t chunkSize, zdnn_ztensor *output) {
+  zdnn_status status;
+  zdnn_tensor_desc *preTransDesc = malloc(sizeof(zdnn_tensor_desc));
```

Comment:
Since you control the free, you could allocate a single `2*sizeof(zdnn_tensor_desc)` array and assign `[0]` to `sizeof(zdnn_tensor_desc)` and `[1]` to `transDesc` and thus have only one malloc.

I would also have an assert / return status. 

Speaking of it, I am confused because many of these functions have return of status but we also assert on failure. This may need to be cleaned up.

Reply:
Good suggestion. Done.